Add chapter2.qmd: fake-data simulation and correlated model setup

Kwan-Jenny opened the pull request:
This adds my Chapter 2 .qmd file.

It includes:
- simulation of fake data with known covariance
- fitting the independence model (baseline)
- setting up the correlated model with Kronecker precision
- wrapper function `run_mod_kron` for running the model

I am mainly stuck on the `prep_priors_multiB()` and `run_mod_kron()` parts, and would like feedback on whether the priors and wrapper are set up correctly.

Kwan-Jenny committed:
# Add chapter2.qmd: fake-data simulation and correlated model setup

Kwan-Jenny committed:
# Update change_log, version #, WORDLIST

d-morrison commented on vignettes/articles/Chapter2.qmd:
```diff
@@ -0,0 +1,611 @@
+
+## 2) SIMULATOR: 
+
+ab_vec <- Vectorize(
+  function(t, y0, y1, t1, alpha, shape)
+    serodynamics:::ab(t, y0, y1, t1, alpha, shape),
+  vectorize.args = c("t","y0","y1","t1","alpha","shape")
+)
```
is this necessary? isn't `serodynamics:::ab()` already vectorized?

d-morrison commented on vignettes/articles/Chapter2.qmd:
```diff
@@ -0,0 +1,611 @@
+)
+
+# Simulator version2: uses serodynamics:::ab 
+simulate_multiB_long2 <- function(
```
please move all function definitions into `R/`, to keep all files short and easy to navigate

d-morrison reviewed (changes requested):
Hi Kwan, sorry to push this review back another day; I'll try to take a detailed look tomorrow. It would really help me if you could move all these functions into the `R/` folder, lint them, and document them with short examples, so I can read and test them individually. My attention span is badly limited right now, and it really helps to have the code in bite-size chunks.

Kwan-Jenny committed:
# Add Kronecker model helper functions and examples

- Added functions: two_phase_y, simulate_multiB_long, simulate_multiB_long2,
  make_ab_vec, prep_priors_multiB, inits_kron, clean_priors,
  write_model_ch2_kron, run_mod_kron
- Added example scripts under inst/examples for each function
- Refactored code from Chapter 2 QMD into R/ with docs + examples

Kwan-Jenny commented:
> > Hi Kwan, sorry to push this review back another day; I'll try to take a detailed look tomorrow. It would really help me if you could move all these functions into the `R/` folder, lint them, and document them with short examples, so I can read and test them individually. My attention span is badly limited right now, and it really helps to have the code in bite-size chunks.
> 
> Hi @d-morrison ,
> 
> I will do exactly as you suggested and let you know once it’s ready, so you can start the review then. I really appreciate you taking the time to review this!

Hi @d-morrison   
I finished moving the functions into `R/`, added docs and short examples, and linted them. It’s ready for you to review now. Thanks again for your help!

Kwan-Jenny committed:
# update

Kwan-Jenny committed:
minor update

Kwan-Jenny committed:
update

d-morrison reviewed (changes requested):
Please finish moving all functions out of the .qmd file and make sure all github checks pass.

Kwan-Jenny committed:
# Remove functions from .qmd (moved to R/ folder) to complete refactor

- Deleted two_phase_y and make_ab_vec functions (and their examples/docs),
  since simulation now directly uses serodynamics:::ab().
- Kept single simulate_multi_b_long() function with serodynamics:::ab()
  as the unified approach.

Kwan-Jenny committed:
WORDLIST update

Kwan-Jenny committed:
minor update

Kwan-Jenny committed:
Update docs with roxygen2 7.3.3

Kwan-Jenny committed:
# minor update

Kwan-Jenny committed:
# update devtools::document()

Kwan-Jenny committed:
# minor update..

Kwan-Jenny committed:
# update...

Kwan-Jenny committed:
# update ....

Kwan-Jenny committed:
# update .....

Kwan-Jenny committed:
# update .....

Kwan-Jenny committed:
# minor update ......

d-morrison commented:
> > Please finish moving all functions out of the .qmd file and make sure all github checks pass.
> 
> Hi @d-morrison , quick clarification before I proceed:
> 
> Do you want me to get all GitHub Actions checks (R-CMD-check on macOS/Windows/Linux, docs, coverage) green before review/approval, or is a clean local `devtools::check()` sufficient?
> 
> Right now `devtools::check()` passes locally, but the PR shows some CI failures. If your expectation is CI-green, I will keep tightening the examples/tests (e.g., wrap JAGS bits in `\dontrun{}`/skip on CI, fix pkgdown) until everything passes.

@Kwan-Jenny please get all the CIs to pass. It's ok to use `\dontrun{}` on the JAGS-dependent examples. For the tests, they should probably pass on at least one operating system, but if you look at the `run_mod()` tests, you'll see that we skip on windows and linux because the results are OS-dependent.

d-morrison commented:
> **Why the docs check failed** (See attached image) <- Questoin here, @d-morrison
> <img alt="image" width="566" height="153" src="https://private-user-images.githubusercontent.com/68584166/494214637-8beb69ea-f2f8-46db-bacd-90d7245f8586.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTg4NTc4MTAsIm5iZiI6MTc1ODg1NzUxMCwicGF0aCI6Ii82ODU4NDE2Ni80OTQyMTQ2MzctOGJlYjY5ZWEtZjJmOC00NmRiLWJhY2QtOTBkNzI0NWY4NTg2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTI2VDAzMzE1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjZmM3OGI4MTYxNjcxNTQ5ZDEzYmI3ODQwZDRhZjIwNmVkMzhmYTE3MzgzMjdjN2MwZDM3NWIwOWQxNTcwYjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6ivbkImNbL1dwG-IcmcHmSBrbcL5isLkr7pPGQAz2K8">

@Kwan-Jenny try updating ggplot2 to v4.0.0 on your computer; github will generally test using the latest released version, and if you look at `autoplot.case_data.R`, you will see:

```r
#' @inheritDotParams ggplot2::geom_point
#' @inheritDotParams ggplot2::geom_line
```

So some documentation will get copied over from ggplot2.

d-morrison commented:
> > **Why the docs check failed** (See attached image) <- Questoin here, @d-morrison
> > <img alt="image" width="566" height="153" src="https://private-user-images.githubusercontent.com/68584166/494214637-8beb69ea-f2f8-46db-bacd-90d7245f8586.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTg4NTc4MTAsIm5iZiI6MTc1ODg1NzUxMCwicGF0aCI6Ii82ODU4NDE2Ni80OTQyMTQ2MzctOGJlYjY5ZWEtZjJmOC00NmRiLWJhY2QtOTBkNzI0NWY4NTg2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTI2VDAzMzE1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjZmM3OGI4MTYxNjcxNTQ5ZDEzYmI3ODQwZDRhZjIwNmVkMzhmYTE3MzgzMjdjN2MwZDM3NWIwOWQxNTcwYjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6ivbkImNbL1dwG-IcmcHmSBrbcL5isLkr7pPGQAz2K8">
> 
> @Kwan-Jenny try updating ggplot2 to v4.0.0 on your computer; github will generally test using the latest released version, and if you look at `autoplot.case_data.R`, you will see:
> 
> ```r
> #' @inheritDotParams ggplot2::geom_point
> #' @inheritDotParams ggplot2::geom_line
> ```
> 
> So some documentation will get copied over from ggplot2.

@Kwan-Jenny I took a closer look at some of the errors you're getting; some of them are snapshots that need to be updated after installing ggplot2 v4.0 on your local computer, and others are due to the `ssdtools` package removing `expect_snapshot_data()` from their package (https://github.com/bcgov/ssdtools/commit/1445f16e4ddad0416639696e968761a6eb23a262). 

I'm working on a fix in #134; keep an eye on that PR, it might help you with this one.

d-morrison commented:
@Kwan-Jenny #134 is now merged; you should resolve conflicts and merge main into this branch, which should clear up some of these CI errors.

Kwan-Jenny committed:
Merge branch 'main' into chapter2-qmd

Kwan-Jenny committed:
Resolve DESCRIPTION/NAMESPACE/WORDLIST

Kwan-Jenny committed:
Merge branch 'chapter2-qmd' of https://github.com/UCD-SERG/serodynamics into chapter2-qmd
pull!

Kwan-Jenny committed:
#update 1 in Oct 1

Kwan-Jenny committed:
# update ggplot2 version to resolve this problem based on Ezra's suggestion

Kwan-Jenny committed:
# minor update in DESCRIPTION

Kwan-Jenny committed:
# fix(simulate_multi_b_long): correct row count

Replace crossing(visit_num,time_days) with visit_num + lookup; rows now n_id*n_blocks*length(time_grid).

Kwan-Jenny committed:
# lint update

Kwan-Jenny committed:
# run_mod_kron: make JAGS runner injectable for tests

- Add `.runjags_fun` arg (defaults to runjags::run.jags).

- Replace direct `runjags::run.jags(...)` call with `.runjags_fun(...)`.

- No runtime change; enables clean mocking in testthat/CI.

Kwan-Jenny committed:
# create 4 test files

Kwan-Jenny committed:
# create simulate_multi_b_long test file

Kwan-Jenny committed:
# lint update on test files

Kwan-Jenny committed:
# run_mod_kron: inject mockable JAGS runner & fix joins

- Add `.runjags_fun` (default `runjags::run.jags`) so tests can fake JAGS.
- Replace brittle joins on attr names with clean lookup tables
  (Subnum→Iso_type, Subject index→Subject_label), then rename to Subject.
- Use `select(-any_of(...))` / `rename(... = any_of(...))` to avoid NSE notes.
- Output schema unchanged; enables CI-safe unit test of run_mod_kron.

Kwan-Jenny committed:
# Preserve `used_priors` in `clean_priors()`; update test

- Keep `used_priors` attribute while dropping legacy fields (`omega`, `wishdf`, `Omega`, `WishDF`, `prec.par`).

- Adjust test to assert attribute preservation and retain existing checks.

Kwan-Jenny committed:
# feat(run_mod): add optional correlated-biomarker mode; remove run_mod_kron

- Add correlated flag to run_mod to toggle Kronecker (Chapter 2) model.
- New args: file_mod_kron (path to CH2 JAGS file).
- When correlated=TRUE:
Clean base priors via clean_priors().
Add Kronecker priors from prep_priors_multi_b() and n_blocks.
Switch inits to inits_kron() and monitor TauB, TauP.
Auto-write fallback model_ch2_kron.jags if missing.

- Preserve original behavior when correlated=FALSE.
- Remove deprecated run_mod_kron(); functionality folded into run_mod.

Kwan-Jenny committed:
Merge branch 'main' into chapter2-qmd

Kwan-Jenny committed:
# update document

Kwan-Jenny committed:
# update format files in package

Kwan-Jenny committed:
# update documentation of run_mod function

Kwan-Jenny committed:
# update lint issue

Kwan-Jenny committed:
# update WORDLIST

Kwan-Jenny committed:
# update init_fun code part

Kwan-Jenny committed:
# update qmd file based on deleting run_mod_kron function

Kwan-Jenny committed:
# Add example code that triggers the JAGS error

Kwan-Jenny committed:
# update documentation

d-morrison committed:
Merge branch 'main' into chapter2-qmd

d-morrison commented on R/Run_Mod.R:
```diff
@@ -26,6 +42,11 @@
 #' Note: These objects can be large.
+#' @param correlated Logical; use Chapter-2 Kronecker prior across biomarkers.
+#'   Default FALSE (independence).
+#' @param file_mod_kron Path to a JAGS file for the Kronecker model.If
```
@Kwan-Jenny please consider this suggestion and explain your decision one way or the other?

d-morrison commented on R/Run_Mod.R:
```diff
@@ -60,18 +81,22 @@
 #'   - An optional `"jags.post"` attribute, included when argument
 #'   `with_post` = TRUE.
 #' @inheritDotParams prep_priors
+#' @seealso clean_priors, prep_priors_multi_b, inits_kron, write_model_ch2_kron
```
please cross-reference wherever possible:
```suggestion
#' @seealso [clean_priors], [prep_priors_multi_b], [inits_kron], [write_model_ch2_kron]
```

d-morrison commented on R/Run_Mod.R:
```diff
@@ -106,8 +131,41 @@ run_mod <- function(data,
+    if (!correlated) {
+      # original (independence) behavior
+      priorspec <- prep_priors(
+        max_antigens = longdata$n_antigen_isos, ...
```
https://style.tidyverse.org/syntax.html#long-function-calls:~:text=If%20a%20function%20call%20is%20too%20long%20to%20fit%20on%20a%20single%20line%2C%20use%20one%20line%20each%20for%20the%20function%20name%2C%20each%20argument%2C%20and%20the%20closing%20)

```suggestion
        max_antigens = longdata$n_antigen_isos, 
        ...
```
